Updated "Using the Microsoft Edge Add-ons API" endpoints (#1694)
MicrosoftDocs/edge-developer: microsoft-edge/extensions-chromium/publish/api/using-addons-api.md

<!-- ====================================================================== -->
## Before you begin

To use the Microsoft Edge Add-ons API, you need to enable the API for your project in the Microsoft Partner Center, by creating API credentials.

> [!NOTE]
> The **Publish APIs** UI is not yet present at Partner Center.

1. Visit Microsoft Partner Center and sign in to the account from which you have already published an add-on.

1. Under the **Microsoft Edge** program, select **Publish API**.

1. In the **Publish API** page, click the **Create API credentials** button.  This step may take a few minutes.

>    :::image type="content" source="../../media/create-api-credentials-button.png" alt-text="The Publish API page at the Microsoft Partner Center showing the Create API credentials button." lightbox="../../media/create-api-credentials-button.png":::

Reviewer: ```suggestion
   :::image type="content" source="../../media/create-api-credentials-button.png" alt-text="The 'Publish API' page at Partner Center after clicking 'Create API credentials', now showing Client ID, Client Secret, and Auth Token URL." lightbox="../../media/create-api-credentials-button.png":::
```

   <!-- make window smaller before capture.  make fresh capture.  save .png in present repo as ../media/create-api-credentials-button.png, replacing path in parens -->

>    The API credentials have now been created; you've enabled or renewed the API.  The **Client ID**, **Client Secret**, and **Auth Token URL** are now displayed on the Publish APIs page.

Reviewer: ```suggestion
   The API credentials have now been created; you've enabled or renewed the API.  The **Client ID**, **Client secret**, **Expiry date**, and **Access token URL** are now displayed on the Publish APIs page.
```

Reviewer: See the image in the rendered page, to check wording about "access" vs. "auth".  @vchapel  
https://review.docs.microsoft.com/en-us/microsoft-edge/extensions-chromium/publish/api/using-addons-api?branch=pr-en-us-1694#before-you-begin

   <!-- possibly add a screenshot here, with fake data with the correct number of characters, or value strings blurred out or erased -->

> 1. Write down the **Client ID**, **Client Secret** and the **Auth Token URL**.  You'll use these values in the next step, to get an access token.

Reviewer: ```suggestion
1. Write down the **Client ID**, **Client secret** and the **Access token URL**.  You'll use these values in the next step, to get an access token.
```

<!-- ====================================================================== -->
## Before you begin

To use the Microsoft Edge Add-ons API, you need to enable the API for your project in the Microsoft Partner Center, by creating API credentials.

> [!NOTE]

> > The **Publish API** page is only present at Partner Center for participants of the private preview.

Reviewer: ```suggestion
> The **Publish API** page is present at Partner Center only for participants of the private preview.
```
The real PR that controls this line is PR 1704.

> 1. Visit Microsoft Partner Center and sign in to the account from which you have already published an add-on.

Reviewer: note fyi: Changed login to sign in per Microsoft Writing Style Guide. https://docs.microsoft.com/en-us/style-guide/a-z-word-list-term-collections/s/sign-in-sign-out

1. Under the **Microsoft Edge** program, select **Publish API**.

1. In the **Publish API** page, click the **Create API credentials** button.  This step may take a few minutes.

   > [!div class="mx-imgBorder"]
   > ![The 'Publish API' page at Partner Center, with a 'Create API credentials' button.](https://user-images.githubusercontent.com/11265822/150777593-42cb9d37-0c1f-4ef4-9de6-9c4bc1166b96.png)

>    <!-- make window smaller before capture.  make fresh capture.  save .png in present repo as ../media/create-api-credentials-button.png, replacing path in parens -->

Reviewer: Re-capture the image and save the resulting .png file with a descriptive filename into the edge-developer repo.
A smaller window before capturing, will produce bigger, more legible text in the rendered article.

To prepare the window to create a screenshot, first resize the Partner Center window as small as possible without breaking the page layout that will result after the credentials are generated and more content is added to the page.

Aim for 1100 px wide, 700 px tall or preferably smaller. In another tab or window, you can go to https://browsersize.com, but don't use that webpage's buttons; just drag the window edge and look at "Browser window size".

   The API credentials have now been created; you've enabled or renewed the API.  The **Client ID**, **Client Secret**, and **Auth Token URL** are now displayed on the Publish APIs page.

>    <!-- possibly add a screenshot here, with fake data with the correct number of characters, or value strings blurred out or erased -->

Reviewer: Possibly add a screenshot here, with fake data with the correct number of characters, or value strings blurred out or erased.

1. Write down the **Client ID**, **Client Secret** and the **Auth Token URL**.  You'll use these values in the next step, to get an access token.

> [!IMPORTANT]

> > Be sure to write down the client secret now, because it's only visible immediately after enabling or renewing the API (that is, after creating API credentials).

Reviewer: @RohitMishra0411 - please check my revised wording in this section.  Read this green text in this view, or to see my exact changes, select Changes from > [my commit; the last item].

<!-- ====================================================================== -->
## Retrieving the access token

> [!NOTE]

> > The Microsoft Edge Add-ons API is currently available only for participants of the private preview.

Reviewer: ```suggestion
> The Microsoft Edge Add-ons API is not yet available for testing.
```

> 1. In the **Publish APIs** page, select the **Create API credentials** button to generate the API credentials.  This step may take a few minutes.  After the APIs are enabled, the **Client ID**, **Client Secret**, and **Auth Token URL** are displayed on this page.

Reviewer: ```suggestion
1. In the **Publish APIs** page, select the **Create API credentials** button to generate the API credentials.  This step may take a few minutes.  After the APIs are enabled, the **Client ID**, **Client Secret**, and **Auth Token URL** are displayed on this page - be ready to note them immediately after the API credentials are generated.
```

> 1. In the **Publish APIs** page, select the **Create API credentials** button to generate the API credentials.  This step may take a few minutes.  After the APIs are enabled, the **Client ID**, **Client Secret**, and **Auth Token URL** are displayed on this page.

Reviewer: Move the image to below this step.

> 1. Note the **ClientID**, **Client Secret** and the **Auth Token URL**. You'll use them in the next step to get an access token. Note that client secret is visible only once, after enabling or renewing the API. 

Reviewer: ```suggestion
1. Note the **ClientID**, **Client Secret** and the **Auth Token URL**. You'll use them in the next step to get an access token. Note that the client secret is visible only once, after enabling or renewing the API. 
```
or: Note that **Client Secret** is visible only once, after enabling or renewing the API.

> 1. Note the **ClientID**, **Client Secret** and the **Auth Token URL**. You'll use them in the next step to get an access token. Note that client secret is visible only once, after enabling or renewing the API. 

Reviewer: Move this step to below the image.

> 1. Note the **ClientID**, **Client Secret** and the **Auth Token URL**. You'll use them in the next step to get an access token. Note that client secret is visible only once, after enabling or renewing the API. 

Reviewer: "Client ID", or "ClientID"?  Two different spellings are shown.

> 

Reviewer: "The client secret is visible only once, after enabling or renewing the API."
Not sure that statement makes sense - or maybe it's clear, in context in the rendered page:
https://review.docs.microsoft.com/en-us/microsoft-edge/extensions-chromium/publish/api/using-addons-api?branch=pr-en-us-1694#before-you-begin
The 2nd part seems like it might be confusing.  Do we need to clarify or strengthen the statement?  After you enable the API or renew the API, then the client secret is shown to you one time only, at that time only?  What can you not do?  Might need to add a concrete, specific "you can't do this" scenario description:
After you enable or renew the API, and do xyz, you can no longer view the client secret.

> ![image](https://user-images.githubusercontent.com/11265822/150777593-42cb9d37-0c1f-4ef4-9de6-9c4bc1166b96.png)

Reviewer: * Should this image be copied into this repo?  
* The alt-text should be descriptive enough to differentiate the image from other images, and should end with a period.
* There should be introductory text or some transition/glue from the lead-in text to the image.  A test is: when reading this .md file without looking at the image & rendered article, a writer should be able to glean what the image shows.

> ![image](https://user-images.githubusercontent.com/11265822/150777593-42cb9d37-0c1f-4ef4-9de6-9c4bc1166b96.png)

Reviewer: ```suggestion
> [!div class="mx-imgBorder"]
> ![alt-text.](https://user-images.githubusercontent.com/11265822/150777593-42cb9d37-0c1f-4ef4-9de6-9c4bc1166b96.png)
```
replace by a .png file in the present repo.

> 

Reviewer: Re-capture the image.  Make the window smaller, like 1100 px wide, 700 px tall or preferably smaller.
In another tab or window, go to https://browsersize.com but don't use that webpage's buttons; just drag the window edge until "Browser window size" is right.
Then make the window you want to capture even smaller, as small as possible without breaking the wrapping/ page layout.  Then capture it.
A smaller window before capturing, will produce bigger, more legible text in the rendered article.

> [!NOTE]
> The Microsoft Edge Add-ons API is not yet available for testing.

> After you've acquired the necessary authorization for your application, get access tokens for APIs.  To get a token using the client credentials grant, send a POST request to the Auth Token URL.  The tenant information is available in the URL that you received in the **Before you begin** steps above.

Reviewer: ```suggestion
After you've acquired the necessary authorization for your application, get access tokens for APIs.  To get a token using the client credentials grant, send a POST request to the Access token URL (the OAuth token).  The tenant information is available in the URL that you received in the **Before you begin** steps above.
```
not sure about this change

Body content: the package file to upload
```

> `$productID` is the ID of the Microsoft Edge Add-on that you want to update. To get the product ID login to Microsoft Partner Center.  Navigate to **Microsoft Edge > Overview**, and then select the extension you want the product ID for.  The extension overview page opens.  The product ID key is available in the extension identity section.

Reviewer 2: ```suggestion
`$productID` is the product ID of the Microsoft Edge Add-on that you want to update. To get the product ID, login to Microsoft Partner Center. Go to **Microsoft Edge** > **Overview**, and then select the extension of the product ID you need.  The extension overview page opens.  The product ID key is available in the **Extension Identity** section.
```

Reviewer 2: I am unable to reproduce the User Interface steps. I have access to Partner Center, but have not used it. If someone can verify the UI steps or take a screen shot of the path or destination page that would be great.

-X PUT \
-T $FILE_NAME \
-v \
https://api.addons.microsoftedge.microsoft.com/v1/products/$productID/submissions/draft/package
```

> If the request succeeded and the update process began, you receive a `202 Accepted` response status code with a `Location` header.  This location header contains the operationID which is required for checking the status of the update operation.

Reviewer 2: ```suggestion
If the request succeeds and the update process begins, you receive a `202 Accepted` response status code with a `Location` header.  This location header contains the `operationID` which is required for checking the status of the update operation.
```

-X POST \
-d "certificationNotes=text value" \
-v \
https://api.addons.microsoftedge.microsoft.com/v1/products/$productID/submissions
```

> If the request succeeded and the publishing process began, you receive a `202 Accepted` response status code with a `Location` header.  This location header contains the operationID which is required for checking the status of the publish operation.

Reviewer 2: ```suggestion
If the request succeeds and the publishing process begins, you receive a `202 Accepted` response status code with a `Location` header.  This location header contains the `operationID` which is required for checking the status of the publish operation.
```
